cvat/apps/engine/serializers.py

```diff
@@ -1207,6 +1206,72 @@ class SubLabeledShapeSerializer(ShapeSerializer, AnnotationSerializer):
 class LabeledShapeSerializer(SubLabeledShapeSerializer):
     elements = SubLabeledShapeSerializer(many=True, required=False)
 
+class LabeledImageSerializerFromDB(serializers.BaseSerializer):
+    def to_representation(self, instance):
+        def convert_tag(tag):
+            attr_keys = ['spec_id', 'value']
+            keys = ['id', 'label_id', 'frame', 'group', 'source']
+
+            result = OrderedDict([(key, tag[key]) for key in keys])
+            result['attributes'] = [
+                OrderedDict([(key, attr[key]) for key in attr_keys]) for attr in tag['labeledimageattributeval_set']
+            ]
+
+            return result
+
+        return convert_tag(instance)
+
+class LabeledShapeSerializerFromDB(serializers.BaseSerializer):
+    def to_representation(self, instance):
+        def convert_shape(shape):
+            attr_keys = ['spec_id', 'value']
+            keys = [
+                'id', 'label_id', 'type', 'frame', 'group',
+                'source', 'occluded', 'outside', 'z_order',
+                'rotation', 'points',
+            ]
+            result = OrderedDict([(key, shape[key]) for key in keys])
+            result['attributes'] = [
+                OrderedDict([(key, attr[key]) for key in attr_keys]) for attr in shape['labeledshapeattributeval_set']
+            ]
+            if shape.get('elements', None) is not None and shape['parent'] is None:
+                result['elements'] = [convert_shape(element) for element in shape['elements']]
+            return result
+
+        return convert_shape(instance)
+
+class LabeledTrackSerializerFromDB(serializers.BaseSerializer):
+    def to_representation(self, instance):
+        def convert_track(track):
+            attr_keys = ['spec_id', 'value']
+            keys = [
+                'id', 'label_id', 'frame', 'group', 'source',
+            ]
+            shape_keys = [
+                'id', 'type', 'frame', 'occluded', 'outside', 'z_order',
+                'rotation', 'points', 'trackedshapeattributeval_set',
+            ]
+            result = OrderedDict([(key, track[key]) for key in keys])
+            result['shapes'] = [
+                OrderedDict([(key, shape[key]) for key in shape_keys]) for shape in track['trackedshape_set']
+            ]
```

Comment:
It would be useful to add a `convert_tracked_shape` function that would do this conversion. That way, all shape-related code would be together (right now, it's split over this and the `for shape` loop below, two discontinuous blocks).

You could also be able to reuse the hypothetical common function that I mentioned in the other comment.